client/descriptions.go

```diff
@@ -245,6 +238,31 @@ func (f FieldKind) String() string {
 	}
 }
 
+// IsObject returns true if this FieldKind is an object type.
+func (f FieldKind) IsObject() bool {
+	return (f == FieldKind_FOREIGN_OBJECT) ||
+		(f == FieldKind_FOREIGN_OBJECT_ARRAY)
```

Comment:
thought: I know this is copy-pasted but I don't think an array of object is an object. One contains the other but the one isn't the other.

Reply:
I'll quickly check if any tests start failing if I change this, otherwise I will leave it as-is for now to limit scope creep.

Although... Having written that, I would probably prefer this function to be removed if it is only checking a single Kind - will see what happens.

- [ ] Have a look, but don't spend too much time

Reply:
Lots of tests start failing, leaving as-is.  If we want to change that it can be done so in another PR.

Reply:
I created an issue for it. https://github.com/sourcenetwork/defradb/issues/2356